ORT import failing for some debian packages

**Describe the bug**
A clear and concise description of what the bug is.

So if a image has the a package like this:
```
- id: deb::passwd:1:4.13+dfsg1-4ubuntu3.2
    purl: pkg:deb/ubuntu/passwd@1:4.13%2Bdfsg1-4ubuntu3.2?arch=amd64
    vcs:
      type:
      url:
      revision:
      path:
    sourceArtifact:
      url:
    declaredLicenses:
      - BSD-3-Clause
    description: |
      change and administer password and group data
       This package includes passwd, chsh, chfn, and many other programs to
       maintain password and group data.
       .
       Shadow passwords are supported.  See /usr/share/doc/passwd/README.Debian
    homepageUrl: https://github.com/shadow-maint/shadow
    authors:
      - Ubuntu Developers
```

the ort import from the package-list will fail because the id is not correctly formatted.
As ORT uses : to separate different part of the identifier, the parts themselves must not contain :. So if any original part does contain :, it needs to be escaped. 

The Error:
```
Caused by: java.lang.IllegalArgumentException: An identifier's properties must not contain ':' because that character is used as a separator in the string representation: type='deb', namespace='', name='passwd', version='1:4.13+dfsg1-4ubuntu3.2'.
```

**System configuration**
- Which version of ScanCode.io are you running? latest
- Are you running the app using Docker? yes
- On which OS? linux runner
- What inputs are you using? container image
- Which pipeline are you running? image pipeline

**To Reproduce**
Steps to reproduce the behavior:
1. Take a image with the above mentioned package
2. Create a package-list
3. Import it into ORT

**Expected behavior**
A clear and concise description of what you expected to happen.
No error

**Screenshots**
If applicable, add screenshots to help explain your problem.

--- a/scanpipe/pipes/ort.py
+++ b/scanpipe/pipes/ort.py
@@ -130,14 +130,28 @@ def get_ort_project_type(project):
         return "docker"
 
 
+def sanitize_id_part(value):
+    """
+    Sanitize an identifier part by replacing colons with underscores.
+    ORT uses colons as separators in the identifier string representation.
+    """
+    if value:
+        return value.replace(":", "_")
+    return value
+
+
 def to_ort_package_list_yml(project):
     """Convert a project object into a YAML string in the ORT package list format."""
     project_type = get_ort_project_type(project)
 
     dependencies = []
     for package in project.discoveredpackages.all():
+        type_ = sanitize_id_part(project_type or package.type)
+        name = sanitize_id_part(package.name)
+        version = sanitize_id_part(package.version)
+
         dependency = Dependency(
-            id=f"{project_type or package.type}::{package.name}:{package.version}",
+            id=f"{type_}::{name}:{version}",
             purl=package.purl,
             sourceArtifact=SourceArtifact(url=package.download_url),
             declaredLicenses=[package.get_declared_license_expression_spdx()],

--- a/scanpipe/tests/pipes/test_ort.py
+++ b/scanpipe/tests/pipes/test_ort.py
@@ -72,3 +72,21 @@ class ScanPipeORTPipesTest(TestCase):
             ],
         }
         self.assertEqual(expected, package_list)
+
+    def test_scanpipe_ort_pipes_to_ort_package_list_yml_sanitization(self):
+        project = make_project(name="Analysis")
+        package_data = {
+            "name": "passwd",
+            "type": "deb",
+            "version": "1:4.13+dfsg1-4ubuntu3.2",
+            "purl": "pkg:deb/ubuntu/passwd@1:4.13%2Bdfsg1-4ubuntu3.2?arch=amd64",
+        }
+        pipes.update_or_create_package(project, package_data)
+
+        package_list_yml = ort.to_ort_package_list_yml(project)
+        package_list = saneyaml.load(package_list_yml)
+        dependency_id = package_list["dependencies"][0]["id"]
+
+        # The colon in the version should be sanitized
+        self.assertNotIn("1:4.13", dependency_id)
+        self.assertEqual("deb::passwd:1_4.13+dfsg1-4ubuntu3.2", dependency_id)

